cupy/cuda/compiler.py

```diff
@@ -263,10 +323,29 @@ def _compile_with_cache_cuda(source, options, arch, cache_dir,
         ptx = compile_using_nvrtc(source, options, arch, name + '.cu')
         ls = function.LinkState()
         ls.add_ptr_data(ptx, 'cupy.ptx')
+        # for separate compilation
+        if _check_cudadevrt_needed(options):
+            global _win32, _osx
+            # defer import to here to avoid circular dependency
+            from cupy.cuda import get_cuda_path
+            cudadevrt = get_cuda_path()
+            if _win32:
+                # rely on os.altsep
+                cudadevrt += '/lib/x64/cudadevrt.lib'
+            elif _osx:
+                cudadevrt += '/lib/libcudadevrt.a'
+            else:  # linux
+                cudadevrt += '/lib64/libcudadevrt.a'
+            cudadevrt = os.path.normpath(cudadevrt)
```

Comment:
Please avoid using `normpath` as CUDA_PATH may contain `..` (it is not very common setup but not explicitly prohibited).

Reply:
I made the requested change, but wouldn't `..` be correctly handled by `normpath`?

Reply:
```python
>>> import os.path
>>> os.path.normpath('/home/leofang/cupy/../')
'/home/leofang'
>>>
```

Reply:
If the parent directory is a symlink, it will be dereferenced by `normpath`.
c.f. https://stackoverflow.com/questions/34865153/os-path-normpath-and-symbolic-links

Reply:
Thanks, @kmaehashi!